tests/integration/devfile/cmd_devfile_list_test.go

```diff
@@ -29,79 +24,104 @@ var _ = Describe("odo list with devfile", func() {
 		helper.CommonAfterEach(commonVar)
 	})
 
-	When("a component created in 'app' application", func() {
+	When("a component created for deployment", func() {
 
 		BeforeEach(func() {
-			helper.Cmd("odo", "create", "--project", commonVar.Project, cmpName, "--devfile", helper.GetExamplePath("source", "devfiles", "nodejs", "devfile.yaml")).ShouldPass()
-			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
-
-		})
-
-		It("should show the component as 'Not Pushed'", func() {
-			output := helper.Cmd("odo", "list").ShouldPass().Out()
-			Expect(helper.Suffocate(output)).To(ContainSubstring(helper.Suffocate(fmt.Sprintf("%s%s%s%sNotPushed", "app", cmpName, commonVar.Project, "nodejs"))))
+			helper.CopyExample(filepath.Join("source", "nodejs"), commonVar.Context)
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-deploy.yaml"), path.Join(commonVar.Context, "devfile.yaml"))
+			helper.Chdir(commonVar.Context)
 		})
 
-		It("should show the component as 'Not Pushed' in JSON output", func() {
-			output := helper.Cmd("odo", "list", "-o", "json").ShouldPass().Out()
-			values := gjson.GetMany(output, "kind", "devfileComponents.0.kind", "devfileComponents.0.metadata.name", "devfileComponents.0.status.state")
-			expected := []string{"List", "Component", cmpName, "Not Pushed"}
-			Expect(helper.GjsonExactMatcher(values, expected)).To(Equal(true))
+		AfterEach(func() {
+			helper.Chdir(commonVar.Context)
```

Comment:
This doesn't make sense, you are already inside the `commonVar.Context` directory. Perhaps you should take a note of what directory you originally started with and then move back to it in `AfterEach`.